### PR TITLE
Refactor pagination component

### PR DESCRIPTION
**Pull request by artursmirnov**
A small refactoring of pagination component, mostly for simplicity and stability. Also extracted from https://github.com/travis-ci/travis-web/pull/1949/

It also fixes the issue with pagination on narrow screens (it wraps and looks broken)

### CODE REVIEW

File: app/components/pagination-navigation.js [hunk 1]
```diff
@@ -1,57 +1,44 @@
...
-      let numberOfPages = this.get('pagination.numberOfPages');
-      let perPage = this.get('pagination.perPage');
-      let currentPage = this.get('pagination.currentPage');
-      let offset = this.get('pagination.offset');
-      let innerWindow = this.get('innerWindow');
-      let outerWindow = this.get('outerWindow');
-
-      let thresholdDisplayAll = ((outerWindow + 1) * 2) + (innerWindow + 1);
+      const { outer, inner, pagination } = this;
+      const { numberOfPages, perPage, currentPage, offset } = pagination || {};
```
Comment (sorta) [inline app/components/pagination-navigation.js]: More condensed and readable. Love it!

File: app/styles/app/modules/pagination.scss [hunk 2]
```diff
@@ -5,7 +5,7 @@
   ul {
     display: flex;
     align-items: baseline;
-    flex-wrap: wrap;
+    flex-wrap: nowrap;
```
Comment (sorta) [inline app/styles/app/modules/pagination.scss]: Hmm yeah the way it was wrapping before didn't look great, but doesn't this cause the outer edges of the pagination to go offscreen when there are a lot of items on a narrow screen? For example https://as-pagination-refactoring.test-deployments.travis-ci.org/travis-ci

Comment (artursmirnov) [reply app/styles/app/modules/pagination.scss]: Hmm, you're right 🤔But I think it should rather collapse than wrap.. I'll think about that.

Comment (artursmirnov) [reply app/styles/app/modules/pagination.scss]: It appears that the best way would be to change `inner` and `outer` parameters depending on window dimensions. But this seems like a significant change, so I'll just revert it to wrap for now and file a separate issue.

Comment (sorta) [reply app/styles/app/modules/pagination.scss]: Cool that sounds good to me. I agree that the current wrapping isn't great and some sort of collapse sounds like a good solution. But yeah that sounds like a significant change, so moving it into a separate issue sounds good 👍 Thanks!

**Pull request merged**